### PR TITLE
Introduced compression levels as an abstraction for the actual algorithm

### PR DESCRIPTION
**Pull request by dgquintas**

### CODE REVIEW

File: src/core/compression/algorithm.c [hunk 1]
```diff
@@ -47,3 +48,21 @@ const char *grpc_compression_algorithm_name(
...
+grpc_compression_algorithm grpc_compression_algorithm_for_level(
+    grpc_compression_level level) {
+  switch (level) {
+    case GRPC_COMPRESS_NONE:
+      return GRPC_COMPRESS_NONE;
+    case GRPC_COMPRESS_LEVEL_LOW:
+    case GRPC_COMPRESS_LEVEL_MED:
+      return GRPC_COMPRESS_DEFLATE;
+    case GRPC_COMPRESS_LEVEL_HIGH:
+      return GRPC_COMPRESS_GZIP;
```
Comment (ctiller) [inline src/core/compression/algorithm.c]: This should be deflate also

Comment (dgquintas) [inline src/core/compression/algorithm.c]: Done

**Comment by dgquintas** [comment]
Done and green.

**Comment by dgquintas** [comment]
Ok, all done.

**Pull request merged**